plugins/kotlin/injection/base/tests/test/org/jetbrains/kotlin/idea/base/injection/KotlinInjectionTestBase.kt

```diff
@@ -330,6 +330,33 @@ abstract class KotlinInjectionTestBase : AbstractInjectionTest() {
         languageId = HTMLLanguage.INSTANCE.id, unInjectShouldBePresent = false
     )
 
+    fun testInjectionOffVarargParameterWithAnnotation() = doInjectionPresentTest(
+        """
+        import org.intellij.lang.annotations.Language
+
+        fun bar(@Language("HTML") vararg s: String){}
+
+        val a = bar(
+        "some","some","<caret>some"
+        )
+        """,
+        languageId = HTMLLanguage.INSTANCE.id, unInjectShouldBePresent = false
+    )
+
+    fun testInjectionOffVarargParameterWithOutAnnotation() = doInjectionPresentTest(
+        """
+        import org.intellij.lang.annotations.Language
+
+        fun bar(vararg s: String,@Language("HTML") s2 : String){}
```

Comment:
for the sake of clearness it would be better to use the same function signature as in previous test
```
fun bar(@Language("HTML") vararg s: String){}
```

Reply:
That is intended to be a different signature. Since this test is for wrong injection of the second vararg with the injection of the 2 argument.

Reply:
I renamed the test case to be more descriptive of what the test should test.